### ISSUE TITLE
onExit exit-handler bug in Sensor Workflow / go-client 'cannot unmarshal object into Go'

### ISSUE THREAD

**Issue by nikopen** [argoproj/argo-events#713]
Argo v2.8.2  
Events v0.16

Running the exit-handler structure, triggered by onExit, specified in a Workflow within a Sensor:

```
  - name: exit-handler
    steps:
      - name: cry
        template: cry
        when: "{{workflow.status}} != Succeeded"
      - name: celebrate
        template: celebrate
        when: "{{workflow.status}} == Succeeded"
  - name: celebrate
    container:
      image: alpine:latest
      command: [sh, -c]
      args: ["curl etc etc"]
  - name: cry
    container:
      image: alpine:latest
      command: [sh, -c]
      args: ["curl etc etc"]
```

The entire workflow will fail to run, the Sensor generates the following error:
```
ERRO[2020-06-23 01:28:31] failed to operate on the event notification   error="v1alpha1.Workflow.Spec: v1alpha1.WorkflowSpec.Templates: []v1alpha1.Template: v1alpha1.Template.Steps: []v1alpha1.ParallelSteps: unmarshalerDecoder: json: cannot unmarshal object into Go value of type []v1alpha1.WorkflowStep, error found in #10 byte of ...|ucceeded\"},{\"contain|..., bigger context ...|\":\"cry\",\"when\":\"{{workflow.status}} != Succeeded\"},{\"container\":{\"args\":\"= curl --silent -X POST -H |..."
```

Something's wrong with the parser when it hits the first 'when' clause, end of line at "Succeeded**"**

The only way I managed to run is by directly specifying the container on `exit-handler` (and having additional "if" on one step).

**Comment by nikopen** [comment]
It seems that there is a double hyphen needed just after "steps" for some reason which I missed from the example....

**Comment by whynowy** [comment]
@nikopen - was the issue resolved? BTW - we had a known issue that some of the new workflow fields were not recognized by `argo-events`, and the issue has been fixed in v0.17.0, I would like you try out the new version and let us know, thanks!

**Comment by whynowy** [comment]
I assume this is not an issue any more, please let us know otherwise.
